torch/csrc/autograd/function.cpp

```diff
@@ -28,4 +28,24 @@ auto Function::flags(const variable_list& inputs) -> FunctionFlags {
   return f;
 }
 
+auto Function::name() -> std::string {
+  return std::string(typeid(*this).name());
+}
+
+auto Function::call_hooks(variable_list inputs) -> variable_list {
+  if (hooks.empty()) {
+    return inputs;
+  }
+
+  variable_list outputs(inputs.size());
+  for (size_t i = 0; i < inputs.size(); ++i) {
+    if (hooks[i]) {
+      outputs[i] = (*hooks[i])(inputs[i]);
```

Comment:
Wouldn't it be better to have either a single hook that gets all outputs (and maybe calls other hooks with some of them), or a number of hooks from all outputs to all outputs?